src/uuid.cr

```diff
@@ -75,22 +127,6 @@ struct UUID
     end
   end
 
-  # Sets UUID variant to specified *value*.
```

Comment:
Why leave out setters? Separate `UUID#variant/version=` methods look cleaner too.

Reply:
Because it's a struct, and mutability is confusing. I don't see any common usecase for those functions. I did however forget to implement `UUID.new(old_uuid, variant: xxx, version: yyy)`.

Reply:
I disagree with this.
I think adding a constructor not only makes simple changes more confusing but also mean more objects will be created for no reason.

Reply:
It's a struct, creating it has practically 0 cost, if not absolutely 0 const if the compiler notices you doing something like `uuid = UUID.new(uuid, variant: foo, version: foo)`. It doesn't make sense to make `UUID` mutable, it's a fixed bit pattern. The version and the variant are not metadata, they're an integral part of the bit pattern. Making a UUID object mutable is like making a string mutable, unpredictable and unnatural.

Reply:
I mean I would not mind just making this a class to get around the struct argument.
Also I do not get why immutability is a big deal?

Can you mutate this through accessing the bytes anyways?

Reply:
Added `UUID.new(UUID, version, variant)`.